### ISSUE TITLE
transformDownloadedImage

### ISSUE THREAD

**Issue by hovven** [rs/SDWebImage#2419]
### New Issue Checklist

* [ ] I have read and understood the [CONTRIBUTING guide](https://github.com/rs/SDWebImage/blob/master/.github/CONTRIBUTING.md)
* [ ] I have read the [Documentation](http://cocoadocs.org/docsets/SDWebImage/)
* [ ] I have searched for a similar issue in the [project](https://github.com/rs/SDWebImage/issues) and found none

### Issue Info

 Info                    | Value                               |
-------------------------|-------------------------------------|
 Platform Name           | e.g. ios / macos / tvos / watchos
 Platform Version        | e.g. 11.0 / 10.13.0 / 11.0 / 4.0
 SDWebImage Version      | e.g. 4.2.0 / 4.1.0
 Integration Method      | e.g. carthage / cocoapods / manually
 Xcode Version           | e.g. Xcode 9 / Xcode 8
 Repro rate              | e.g. all the time (100%) / sometimes x% / only once
 Repro with our demo prj | e.g. does it happen with our demo project?
 Demo project link       | e.g. link to a demo project that highlights the issue

### Issue Description and Steps

Please fill in the detailed description of the issue (full output of any stack trace, compiler error, ...) and the steps to reproduce the issue.

what am i doing wrong?
https://stackoverflow.com/questions/51635633/resize-image-with-sdwebimage-in-uitableview

**Comment by dreampiggy** [comment]
In 4.x, there is a delegate method, for you to do a transformation after image was downloaded, the method is called in the global queue. See [imageManager:transformDownloadedImage:withURL:](https://github.com/rs/SDWebImage/blob/master/SDWebImage/SDWebImageManager.h#L156-L166)


```objective-c
SDWebImageManager *manager = SDWebImageManager.sharedManager;
manager.delegate = self;

- (nullable UIImage *)imageManager:(nonnull SDWebImageManager *)imageManager transformDownloadedImage:(nullable UIImage *)image withURL:(nullable NSURL *)imageURL {
    CGSize size = CGSizeMake(100, 100);
    return [image sd_resizedImageWithSize:size]; // Implement your own, such as writing a category on UIImage
}
```

In 5.x, you can simply use `SDWebImageResizeTransformer` instead, one line code is enough. See [Wiki - Image Transformer](https://github.com/rs/SDWebImage/wiki/Advanced-Usage#image-transformer-50) 

```objective-c
SDWebImageManager *manager = SDWebImageManager.sharedManager;
CGSize size = CGSizeMake(100, 100);
manager.transformer = [SDImageResizingTransformer transformerWithSize:size scaleMode:0];
```

**Comment by bpoplauschi** [comment]
@hovven Thanks for taking interest in our project and reporting this issue. 
I have re-added the template info to the description of this issue. Please fill in this info, it is important for us to know this.

**Comment by bpoplauschi** [comment]
Closing due to lack of response. @hovven feel free to reopen.
